crates/sargon/src/radix_connect/wallet_interaction/dapp_wallet_interaction/wallet_to_dapp/success_response/pre_authorization/subintent_response.rs

```diff
@@ -29,11 +29,18 @@ impl WalletToDappInteractionSubintentResponseItem {
     fn subintent_hash(&self) -> SubintentHash {
         self.signed_subintent.subintent.hash()
     }
+}
 
-    fn expiration_timestamp_seconds(&self) -> u64 {
-        self.expiration_timestamp
-            .duration_since(Timestamp::UNIX_EPOCH)
-            .as_seconds_f64() as u64
+impl From<SignedSubintent> for WalletToDappInteractionSubintentResponseItem {
+    fn from(value: SignedSubintent) -> Self {
+        let expiration_timestamp = value
+            .subintent
+            .header
+            .max_proposer_timestamp_exclusive
+            .expect(
+                "A SignedSubintent must have max_proposer_timestamp_exclusive",
+            );
```

Comment:
As discussed on meeting, added this force cast to avoid adding defensive programming against our own code. Otherwise, we can also make this a `TryFrom` and handle/ignore the error on host apps.

@GhenadieVP